### PR TITLE
blockchain_utilities: simplify getting block blob from height

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/blockchain_utilities/blockchain_ancestry.cpp [hunk 1]
```diff
@@ -482,8 +481,7 @@ int main(int argc, char* argv[])
                 }
                 else
                 {
-                  const crypto::hash block_hash = db->get_block_hash_from_height(od.height);
-                  cryptonote::blobdata bd = db->get_block_blob(block_hash);
+                  cryptonote::blobdata bd = db->get_block_blob_from_height(od.height);
```
Comment (xiphon) [inline src/blockchain_utilities/blockchain_ancestry.cpp]:
```suggestion
                  const cryptonote::blobdata bd = db->get_block_blob_from_height(od.height);
```

Comment (moneromooo-monero) [reply src/blockchain_utilities/blockchain_ancestry.cpp]: Since this isn't a ref or pointer, there's not much point.

Comment (xiphon) [reply src/blockchain_utilities/blockchain_ancestry.cpp]: Is there any point to have it non const?

**Comment by hyc** [comment]
There's a comment that's been copy/pasted into several of these files:

````
blockchain_ancestry.cpp:  // 2. exporter needs to use get_current_blockchain_height(), get_block_id_by_height(), get_block_by_hash()
blockchain_depth.cpp:  // 2. exporter needs to use get_current_blockchain_height(), get_block_id_by_height(), get_block_by_hash()
blockchain_export.cpp:  // 2. exporter needs to use get_current_blockchain_height(), get_block_id_by_height(), get_block_by_hash()
blockchain_usage.cpp:  // 2. exporter needs to use get_current_blockchain_height(), get_block_id_by_height(), get_block_by_hash()
````

Which I'm not sure is correct any more. ?

File: src/blockchain_utilities/blockchain_ancestry.cpp (hunk 1 above)
Comment (moneromooo-monero) [reply src/blockchain_utilities/blockchain_ancestry.cpp]: No.

**Comment by moneromooo-monero** [comment]
It seems at least some of those don't need high level functions indeed.

**Pull request merged**